subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -132,6 +138,87 @@ static void azure_iot_hub_notify_event(struct azure_iot_hub_evt *evt)
 	}
 }
 
+const char *state_name_get(enum connection_state state)
+{
+	switch (state) {
+	case STATE_IDLE: return "STATE_IDLE";
+	case STATE_INIT: return "STATE_INIT";
+	case STATE_CONNECTING: return "STATE_CONNECTING";
+	case STATE_CONNECTED: return "STATE_CONNECTED";
+	case STATE_DISCONNECTING: return "STATE_DISCONNECTING";
+	default: return "STATE_UNKNOWN";
+	}
+}
+
+static void connection_state_set(enum connection_state new_state)
+{
+	bool notify_error = false;
+
+	/* Check for legal state transitions. */
+	switch (connection_state) {
+	case STATE_IDLE:
+		if (new_state != STATE_INIT) {
+			notify_error = true;
+		}
+		break;
+	case STATE_INIT:
+		if (!(new_state != STATE_CONNECTING ||
```

Comment:
I don't think this check is correct, note that `(new_state != STATE_CONNECTING || new_state != STATE_INIT)` will always evaluate to `true`.

Shouldn't this rather be:
```
if (new_state != STATE_CONNECTING && 
    new_state != STATE_IDLE) {
}
```
? Same would apply to other checks below.

Reply:
Woops. You are right. Glad we sorted that one out.